File: src/main/java/sprykerFileCreator/model/definition/spryker/YvesPluginDefinition.java
package pav.sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class YvesPluginDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.YVES_PLUGIN;
    }

    @Override
    public String getNamePattern() {
        return "{{ className }}Plugin";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Yves\\{{ bundleName }}\\Plugin";
    }

    @Override
    public String getMethodForReturnType() {
        return "";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {
                SprykerConstants.YVES_FACTORY,
                SprykerConstants.CLIENT,
        };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Yves\\Kernel\\AbstractPlugin";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/YvesPluginDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class YvesPluginDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.YVES_PLUGIN
    override val namePattern: String
        get() = "{{ className }}Plugin"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Yves\\{{ bundleName }}\\Plugin"
    override val methodForReturnType: String
        get() = ""
    override val docBlockClasses: Array<String>
        get() = arrayOf(
                SprykerConstants.YVES_FACTORY,
                SprykerConstants.CLIENT)
    override val defaultParentClass: String
        get() = "Spryker\\Yves\\Kernel\\AbstractPlugin"
}